tests/foreman/api/test_syncplan.py

```diff
@@ -1100,21 +1100,13 @@ def test_positive_synchronize_custom_product_weekly_recurrence(self):
         ).create()
         sync_plan.add_products(data={'product_ids': [product.id]})
         # Verify product is not synced and doesn't have any content
-        self.logger.info('Waiting {0} seconds to check product {1}'
-                         ' was not synced'.format(delay/4, product.name))
-        sleep(delay/4)
```

Comment:
That is not good approach as you ruin initial idea of the test, but if so many people think otherwise...

Reply:
Could you try to make that comment a little more actionable? Saying "that is not good" does not really help anyone to improve.

Could you re-state your comment in relation to [proposed scenario](https://github.com/SatelliteQE/robottelo/pull/6793#issuecomment-471004235)? Is this proposal lacking something? What? Why?

Or does the current implementation not follow stated proposal? What have we missed?

Reply:
You can form your statements in one question - "Can you be more precise?". Not sure why you put explanation like I am in kindergarten.

First:
'Environments that are very slow or under heavy load' - is definitely not an argument and such parameters can be taken into account in longer delays or better conditions.

Second:
Explanation is very easy. Previous scenario verified that sync plan do not start right away or in 1/4 or in 3/4 time term, so conditions you put in sync plan (the whole idea of test case) is actually working, but now I have no idea at all what do you verify by that test case

Reply:
It should be like

start date = March 7 (current_day = March 14 - 1 week) + 4 minutes
so next occurrence should happen in 4 minutes, not earlier

Reply:
> Previous scenario verified that sync plan do not start right away

We still check that. Lines 1103-1106 are responsible for that.

> or in 3/4 time term

That's not true. This test always checked that product is synced only once, in about 1/4 of delay. 3/4 delay only gave time for sync plan to run, but we never checked if sync started after whole `delay`, or after 1/2 of delay, or anywhere in between.

> so next occurrence should happen in 4 minutes, not earlier
> so you check that product not synced right now, then in 60 seconds and then in 120 seconds.

That sounds like a good approach. Too bad we learn about it only now, when this is already merged.

Could you file a new issue for that? Please mention @swadeley in it, so he is in the loop of further developments or can find it once he has time to work on this. Thanks.

Reply:
created https://github.com/SatelliteQE/robottelo/pull/6793